### PR TITLE
Do not suggest super methods

### LINKED ISSUE #419: FR: Provide no suggestions for `__construct`, `__wakeup`...
Because it is not intended to do so, but psysh still suggests its usage. :/

### PR DESCRIPTION
**Pull request by jens1o**
fixes #419 

#420

### CODE REVIEW

**Comment by jens1o** [comment]
yeeeyy, it works 😃

File: src/Psy/TabCompletion/Matcher/ObjectMethodsMatcher.php [hunk 1]
```diff
@@ -50,7 +50,10 @@ public function getMatches(array $tokens, array $info = array())
         return array_filter(
             get_class_methods($object),
             function ($var) use ($input) {
-                return AbstractMatcher::startsWith($input, $var);
+                return
```
Comment (bobthecow) [inline src/Psy/TabCompletion/Matcher/ObjectMethodsMatcher.php]: I know it's completely arbitrary, and the linter didn't even complain, but for a multiline return I prefer to wrap it in parens (as if it were a multi line array).

Comment (jens1o) [reply src/Psy/TabCompletion/Matcher/ObjectMethodsMatcher.php]: Okay, will do that this afternoon.

Comment (jens1o) [reply src/Psy/TabCompletion/Matcher/ObjectMethodsMatcher.php]:
```php

                return
                    (AbstractMatcher::startsWith($input, $var)
                    // also check that we do not suggest invoking a super method(__construct, __wakeup, …)
                    && !AbstractMatcher::startsWith('__', $var));
```

Do you expect it like this?

Comment (bobthecow) [reply src/Psy/TabCompletion/Matcher/ObjectMethodsMatcher.php]:
```php
return (
    AbstractMatcher::startsWith($input, $var) &&
    // also check that we do not suggest invoking a super method(__construct, __wakeup, …)
    !AbstractMatcher::startsWith('__', $var)
);
```

**Comment by jens1o** [comment]
Okay, now it complies with the code style :/

**Comment by bobthecow** [comment]
Gah. Sorry. 

Okay, do…

```
return foo &&
    bar;
```

?

**Comment by jens1o** [comment]
No worries, just let's hope together.

**Comment by bobthecow** [comment]
Thanks!

**Pull request merged**